pinot-core/src/main/java/com/linkedin/pinot/core/query/scheduler/SchedulerGroupAccountant.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.query.scheduler;
+
+/**
+ * Accounting interface to track resource utilization by a scheduler group.
+ * Scheduler group represents a subqueue if a typical multi-level scheduler.
+ * Sub-queues can selected using SchedulerGroupMapper.
+ *
+ * We mainly account for wall clock time of each thread for a query. This captures
+ * CPU and IO cost for each query but also penalizes for GC activity. We do not account
+ * for memory utilization yet. Nevertheless, wall clock time is a good approximation for
+ * resource utilization in an online system.
+ */
+public interface SchedulerGroupAccountant extends Comparable<SchedulerGroupAccountant> {
```

Comment:
Check BoundedAccountingExecutor. We pass SchedulerGroupAccountant that limits the possible operations(safety) and if required we can have simpler implementations - see comment above that schedulerQ need not be SchedulerGroupBased (our existing algorithm does not have scheduler groups)